node/network/dispute-distribution/src/receiver/mod.rs

```diff
@@ -191,45 +206,35 @@ where
 
 	/// Actual work happening here.
 	async fn run_inner(&mut self) -> Result<()> {
-		let msg = MuxedMessage::receive(&mut self.pending_imports, &mut self.receiver).await?;
-
-		let incoming = match msg {
-			// We need to clean up futures, to make sure responses are sent:
-			MuxedMessage::ConfirmedImport(m_bad) => {
-				self.ban_bad_peer(m_bad)?;
-				return Ok(())
-			},
-			MuxedMessage::NewRequest(req) => req,
+		let msg = if self.peer_queues.is_empty() {
+			// No point to wake on timeout:
+			Some(MuxedMessage::receive(&mut self.pending_imports, &mut self.receiver).await?)
+		} else {
+			self.wait_for_message_or_timeout().await?
 		};
 
-		self.metrics.on_received_request();
-
-		let peer = incoming.peer;
-
-		// Only accept messages from validators:
-		if self.authority_discovery.get_authority_ids_by_peer_id(peer).await.is_none() {
-			incoming
-				.send_outgoing_response(OutgoingResponse {
-					result: Err(()),
-					reputation_changes: vec![COST_NOT_A_VALIDATOR],
-					sent_feedback: None,
-				})
-				.map_err(|_| JfyiError::SendResponse(peer))?;
-
-			return Err(JfyiError::NotAValidator(peer).into())
-		}
+		if let Some(msg) = msg {
+			let incoming = match msg {
+				// We need to clean up futures, to make sure responses are sent:
+				MuxedMessage::ConfirmedImport(m_bad) => {
```

Comment:
Do we know that the import is going to be bad before passing the result into ban_bad_peer? If not, then maybe the naming of m_bad could be changed to reflect that we don't know whether the import is bad yet. Perhaps consider renaming ban_bad_peer to ban_peer_if_bad too. 

I might be missing why these original names are appropriate, so feel free to disregard!

Reply:
Good point. m_bad is meant to mean `maybe_bad` :see_no_evil: - could definitely be more clear! Thanks for the review!

Reply:
Did not survive the rewrite.